docs/guides/search-indexing.rst

```diff
@@ -0,0 +1,322 @@
+in replace of the default search engine of your site.
+To accomplish this, Read the Docs parses the content directly from your HTML pages [*]_.
+
+If you are the author of a theme or a static site generator you can read the next document,
+and follow some conventions in order to improve the integration of SSS with your theme/site.
+
+Indexing
+--------
+
+The content of the page is parsed into sections,
+in general, the indexing process happens in three steps:
+
+#. Identify the main content node.
+#. Remove any irrelevant content from the main node.
+#. Parse all sections inside the main node.
+
+Read the Docs makes use of ARIA_ roles and other heuristics in order to process the content.
+
+.. tip::
+
+   Following the ARIA_ conventions will also improve the accessibility of your site.
+   See also https://webaim.org/techniques/semanticstructure/.
+
+.. _ARIA: https://www.w3.org/TR/wai-aria/
+
+Main content node
+~~~~~~~~~~~~~~~~~
+
+The main content node should have a main role (or a ``main`` tag), and should exists only one per page.
+This node is the one that contains all the page content. Example:
+
+.. code-block:: html
+   :emphasize-lines: 10-12
+
+   <html>
+      <head>
+         ...
+      </head>
+      <body>
+         <div>
+            This content isn't processed
+         </div>
+
+         <div role="main">
+            All content inside the main node is processed
+         </div>
+
+         <footer>
+            This content isn't processed
+         </footer>
+      </body>
+   </html>
+
+Irrelevant content
+~~~~~~~~~~~~~~~~~~
+
+If you have content inside the main node that isn't relevant to the page
+(like navigation items, menus, or search box),
+make sure to use the correct role or tag for it.
+
+Roles to be ignored:
+
+- ``navigation``
+- ``search``
+
+Tags to be ignored:
+
+- ``nav``
+
+Example:
+
+.. code-block:: html
+   :emphasize-lines: 3-5
+
+   <div role="main">
+      ...
+      <nav role="navigation">
+         ...
+      </nav>
+      ...
+   </div>
+
+Sections
+~~~~~~~~
+
+Sections are ``h`` tags, and sections of the same level should be neighbors.
+Additionally, sections should have an unique ``id`` attribute per page (this is used to link to the section).
+All content bellow the section, till the new section will be indexed as part of the section. Example:
+
+.. code-block:: html
+   :emphasize-lines: 2-10
+
+   <div role="main">
+      <h1 id="section-title">
+         Section title
+      </h1>
+      <p>
+         Content to be indexed
+      </p>
+      <ul>
+         <li>This is also part of the section and will be indexed as well</li>
+      </ul>
+
+      <h2 id="2">
+         This is the start of a new section
+      </h2>
+      <p>
+         ...
+      </p>
+
+      ...
+
+      <h1 id="neigbor-section">
+         This section is neighbor of "section-title"
+      </h1>
+      <p>
+         ...
+      </p>
+   </div>
+
+Sections can also be wrapped till two levels (and have nested sections),
```

Comment:
I don't understand this line "wrapped till two levels"

Reply:
What about

> Sections can be inside till two nested tags (and have nested sections),